File: Python/Containers/MongoDB-REST/src/health/health_results.py
import time
from typing import List

from .health_result_entry import HealthResultEntry

class HealthResults:
    def __init__(self, duration: int, entries: List[HealthResultEntry]):
        status = 'healthy'
        healthy = True
        for entry in entries:
            if not entry.healthy:
                status = 'unhealthy'
                healthy = False
                break

        self.duration = f'{duration} ms'
        self.healthy = healthy
        self.status = status
        self.results = entries
        self.timestamp = time.strftime('%Y-%m-%dT%H:%M:%SZ', time.gmtime())
        self.version = '1.0.0'

    def to_dict(self):
        return {
            "duration": self.duration,
            "healthy": self.healthy,
            "status": self.status,
            "results": [entry.to_dict() for entry in self.results],
            "timestamp": self.timestamp,
            "version": self.version
        }

File: Examples/Rust/Containers/MongoDB-REST/src/src/health/health_results.rs
use serde::{Serialize, Deserialize};

use crate::health::health_result_entry::HealthResultEntry;

#[derive(Serialize, Deserialize)]
pub struct HealthResults {
    pub r#type: String,
    pub duration: String,
    pub healthy: bool,
    pub status: String,
    pub results: Vec<HealthResultEntry>,
    pub timestamp: String,
    pub version: String,
}

impl HealthResults {
    pub fn new(duration: u64, entries: Vec<HealthResultEntry>) -> Self {
        let mut status = "healthy".to_string();
        let mut healthy = true;
        for entry in &entries {
            if !entry.healthy {
                status = "unhealthy".to_string();
                healthy = false;
                break;
            }
        }
        HealthResults {
            r#type: "rust".to_string(),
            duration: format!("{} ms", duration),
            healthy,
            status,
            results: entries,
            timestamp: chrono::Utc::now().to_rfc3339(),
            version: "1.0.0".to_string(),
        }
    }
}